### ISSUE TITLE
Can't disable INTEGRATION_TO_CALLBACK

### ISSUE THREAD

**Issue by LZY-the-boys** [huggingface/transformers#18878]
### System Info

I use transformers.4.20.0

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

When using the huggingface Trainer,  I set `report_to=['none'] ` in training_args to disable wandb logging as the doc says but a value error will be raised. I notice it's because the problem in the following code of `transformers/integrations.py`:

```python
def get_reporting_integration_callbacks(report_to):
    for integration in report_to:
        if integration not in INTEGRATION_TO_CALLBACK:
            raise ValueError(
                f"{integration} is not supported, only {', '.join(INTEGRATION_TO_CALLBACK.keys())} are supported."
            )
    return [INTEGRATION_TO_CALLBACK[integration] for integration in report_to]
```
No 'none' logits is dealed with, nor any other disable methods?  

### Expected behavior

I don't know if it's expected, but it's confusing me and causing inconvenience. So i would like to get the answer :)

**Comment by LysandreJik** [comment]
cc @sgugger

**Comment by sgugger** [comment]
No, this code is not called when you set `report_to="none"` or `report_to=["none"]` since it is filtered out [here](https://github.com/huggingface/transformers/blob/6678350c01629b848aa9c41e169da5d6b8d9e7e9/src/transformers/training_args.py#L1155).

Setting `report_to=["none"]` in `TrainingArugments` works perfectly on my end, please make sure you are using the latest Transformers version and if the issue persists, please give us a code reproducer.
